lib_eio_linux/eio_stubs.c

```diff
@@ -148,24 +152,62 @@ CAMLprim value caml_eio_pidfd_send_signal(value v_pidfd, value v_signal) {
   CAMLreturn(Val_unit);
 }
 
+static int pidfd_open(pid_t pid, unsigned int flags) {
+  return syscall(SYS_pidfd_open, pid, flags);
+}
+
+/* Like clone3, but falls back to fork if not supported.
+   Also, raises exceptions rather then returning an error. */
+static pid_t clone3_with_fallback(struct clone_args *cl_args) {
+  int *pidfd = (int *)(uintptr_t) cl_args->pidfd;
+  pid_t child_pid = syscall(SYS_clone3, cl_args, sizeof(struct clone_args));
+
+  if (child_pid >= 0)
+    return child_pid;		/* Success! */
+
+  if (errno != ENOSYS && errno != EPERM) {
+    uerror("clone3", Nothing);	/* Unknown error */
```

Comment:
This needs to be fixed everywhere, not just here, but I believe the recommended function from OCaml 5 is `caml_uerror` so it's namespaced right. (see ocaml/ocaml@67a4d75cfafa040099cdd322e23464362359af29).

Reply:
OK, let's do that in a separate PR.